src/jit/instr.cpp

```diff
@@ -239,13 +239,18 @@ void CodeGen::instNop(unsigned size)
 void CodeGen::inst_JMP(emitJumpKind jmp, BasicBlock* tgtBlock)
 {
 #if !FEATURE_FIXED_OUT_ARGS
-    // On the x86 we are pushing (and changing the stack level), but on x64 and other archs we have
-    // a fixed outgoing args area that we store into and we never change the stack level when calling methods.
-    //
-    // Thus only on x86 do we need to assert that the stack level at the target block matches the current stack level.
-    //
-    assert(tgtBlock->bbTgtStkDepth * sizeof(int) == genStackLevel || compiler->rpFrameType != FT_ESP_FRAME);
+// On the x86 we are pushing (and changing the stack level), but on x64 and other archs we have
+// a fixed outgoing args area that we store into and we never change the stack level when calling methods.
+//
+// Thus only on x86 do we need to assert that the stack level at the target block matches the current stack level.
+//
+#ifndef UNIX_X86_ABI
+    assert(tgtBlock->bbTgtStkDepth * sizeof(int) == genStackLevel || isFramePointerUsed());
+#else
+    // bbTgtStkDepth is a (pure) argument count (stack alignment padding should be excluded).
+    assert(tgtBlock->bbTgtStkDepth * sizeof(int) == genStackLevel - curNestedAlignment || isFramePointerUsed());
```

Comment:
@BruceForstall I'll add CLANG_FORMAT_COMMENT_ANCHOR; to make comment formatting better, but it seems that argStackLevel  will be unused for Release build (Please correct me if there is something wrong).